examples/stake-tokens/components/types.tsx

```diff
@@ -0,0 +1,171 @@
+  isLoading?: boolean;
+  isDisabled?: boolean;
+  icon?: IconType;
+  onClickConnectBtn?: MouseEventHandler<HTMLButtonElement>;
+}
+
+export interface ConnectedUserCardType {
+  walletIcon?: string;
+  username?: string;
+  icon?: ReactNode;
+}
+
+export interface FeatureProps {
+  title: string;
+  text: string;
+  href: string;
+}
+
+export interface ChainCardProps {
+  prettyName: string;
+  icon?: string;
+}
+
+export type CopyAddressType = {
+  address?: string;
+  walletIcon?: string;
+  isLoading?: boolean;
+  maxDisplayLength?: number;
+  isRound?: boolean;
+  size?: string;
+};
+
+export interface OptionBase {
+  variant?: string;
+  colorScheme?: string;
+  isFixed?: boolean;
+  isDisabled?: boolean;
+}
+
+export interface ChainOption extends OptionBase {
+  isDisabled?: boolean;
+  label: string;
+  value: string;
+  icon?: string;
+  chainName: string;
+  chainRoute?: string;
+}
+
+export type handleSelectChainDropdown = (value: ChainOption | null) => void;
+
+export interface ChangeChainDropdownType {
+  data: ChainOption[];
+  selectedItem?: ChainOption;
+  onChange: handleSelectChainDropdown;
+  chainDropdownLoading?: boolean;
+}
+
+export interface ChangeChainMenuType {
+  data: ChainOption[];
+  value?: ChainOption;
+  onClose?: () => void;
+  onChange: handleSelectChainDropdown;
+  innerRef?: RefObject<HTMLInputElement>;
+}
+
+export interface Any {
+  typeUrl: string;
+  value: Uint8Array;
+}
+
+export enum BondStatus {
+  BOND_STATUS_UNSPECIFIED = 0,
+  BOND_STATUS_UNBONDED = 1,
+  BOND_STATUS_UNBONDING = 2,
+  BOND_STATUS_BONDED = 3,
+  UNRECOGNIZED = -1,
+}
+
+export interface Description {
+  moniker: string;
+  identity: string;
+  website: string;
+  securityContact: string;
+  details: string;
+}
+
+export interface CommissionRates {
+  rate: string;
+  maxRate: string;
+  maxChangeRate: string;
+}
+
+export interface Commission {
+  commissionRates?: CommissionRates | undefined;
+  updateTime?: Date | undefined;
+}
+
+export interface Validator {
+  operatorAddress: string;
+  consensusPubkey?: Any | undefined;
+  jailed: boolean;
+  status: BondStatus;
+  tokens: string;
+  delegatorShares: string;
+  description?: Description | undefined;
+  unbondingHeight: Long;
+  unbondingTime?: Date | undefined;
+  commission?: Commission | undefined;
+  minSelfDelegation: string;
+}
+
+export interface MyValidator {
+  details: string | undefined;
+  name: string | undefined;
+  address: string;
+  staked: number;
+  reward: number;
+}
+
+export interface Delegation {
+  delegatorAddress: string;
+  validatorAddress: string;
+  shares: string;
+}
+
+export interface Coin {
+  denom: string;
+  amount: string;
+}
+
+export interface DelegationResponse {
+  delegation?: Delegation | undefined;
+  balance?: Coin | undefined;
+}
+
+export interface DecCoin {
+  denom: string;
+  amount: string;
+}
+
+export interface Reward {
+  validatorAddress: string;
+  reward: DecCoin[];
+}
+
+export interface DelegationTotalRewards {
```

Comment:
so let's delete all of 
- [ ] `DelegationTotalRewards`
- [ ] `Reward`
- [ ] `DecCoin`
- [ ] `DelegationResponse`
- [ ] `Coin`
- [ ] `Delegation`
- [ ]  (anything representing a cosmos-based type)

Reply:
if you want to import the types, I'm happy to help, here is an example I've done to pull in types if you want to use `import type`, but I think you may be able to simply pull them off of `cosmos.bank.v1beta1` and `cosmos.distribution.v1beta1`

```
import type { MsgCreateGroupWithPolicyEncoded } from 'interchain46/types/codegen/cosmos/group/v1/tx'
```